Bench 105 create docker image

PietroConvalleAD opened the pull request:
Adds DockerFile and docker-compose.yml

Commit:
created docker file and updated docker-compose

Commit:
minor changed, added some information to the ReadMe

Commit:
fixed port error

Commit:
removed not used args from DockerFile

Commit:
added branch argument in docker file

Commit:
removed file .env from docker compose and added explicit environtment

Commit:
added publish workflow

Commit:
moved entrypoint in DockerFile

Commit:
updated README

Commit:
test package workflow

Commit:
added workflow dispatch

Commit:
renamed workflow file

Commit:
added workflow on_pull_request

Commit:
trigger workflows

Commit:
added dockerfile name

Commit:
added image field

Commit:
added migrations and timezone dependencies

JoeCSykes committed:
condensed DockerFile

JoeCSykes committed:
testing docker workflow

JoeCSykes committed:
trigger on pull request

JoeCSykes committed:
changed name of docker file

PietroConvalleAD commented on pyproject.toml:
```diff
@@ -19,13 +19,13 @@ drf-problems = { git = "https://github.com/Axeltherabbit/drf-problems" }
 pyright = "^1.1.278"
-coverage = "^6.5.0"
 pycodestyle = "^2.9.1"
-drf-yasg = "^1.21.4"
```
why were these moved to main dependencies?

PietroConvalleAD commented on pyproject.toml:
```diff
@@ -17,13 +17,15 @@ django-json-404-middleware = { git = "https://github.com/Axeltherabbit/django-js
+tzdata = "^2022.7"
+coverage = "^6.5.0"
+drf-yasg = "^1.21.4"
 
 [tool.poetry.group.dev.dependencies]
 black = "^22.10.0"
 pyright = "^1.1.278"
-coverage = "^6.5.0"
 pycodestyle = "^2.9.1"
-drf-yasg = "^1.21.4"
```
why were these moved to main dependencies?

JoeCSykes replied:
because it was erroring saying the coverage module and the drf-yasg modules were not installed. They weren't being installed because they were in the dev dependencies and you were running poetry install --only-main. As such I moved them into main so it didn't error

PietroConvalleAD replied:
of course, you have to set your .env to use the .base settings.py.

JoeCSykes replied:
I don't understand. this seems a lot more complicated than just including them in the main dependencies

PietroConvalleAD replied:
uh? no, it's what DJANGO_SETTINGS_MODULE= in the .env file was made for

PietroConvalleAD replied:
`DJANGO_SETTINGS_MODULE=answerking.settings.base`

PietroConvalleAD commented on Dockerfile:
```diff
@@ -0,0 +1,13 @@
+
+CMD [ "/usr/local/bin/poetry", "run", "python", "manage.py", "migrate", "0.0.0.0:8000"]
+CMD [ "/usr/local/bin/poetry", "run", "python", "manage.py", "runserver", "0.0.0.0:8000"]
+EXPOSE 8000
```
Expose is defined in docker-compose

JoeCSykes committed:
added wait func to ensure db is up before running app

JoeCSykes committed:
corrected so can send requests + updated ReadMe details

PietroConvalleAD commented on .env.production:
```diff
@@ -0,0 +1,8 @@
+DATABASE_NAME=answerking_app
+DATABASE_HOST=host.docker.internal
+DATABASE_PORT=3306
+DATABASE_USER=root
+DATABASE_PASS=Admin123
+SECRET_KEY="django-insecure-x977=v5a2q-e%_5$b9ge@jzk%s_nhk2l8_use&h@*m$w33dopr"
+DJANGO_SETTINGS_MODULE=answerking.settings.base
+DATABASE_ENGINE = "django.db.backends.mysql"
```
why? use github secrets as you'd normally do, you might have to override it in the workflow using another variable `DJANGO_SETTINGS_MODULE = {{GITHUB.DJANGO_SETTINGS_MODULE_DEPLOY}}`

JoeCSykes replied:
this si when you are running it locally so it wouldn't have access to github secrets

JoeCSykes committed:
run make prepare + changed git ignore to ignore any file starting with .env

JoeCSykes committed:
Delete .env.production

PietroConvalleAD commented:
Does build now work if you switch the merge to release?

JoeCSykes commented:
no it should run when someone pushes to a branch named release-*

PietroConvalleAD commented:
Yeah I mean, just switch it to release-test to test it maybe

JoeCSykes commented:
Created a release-test branch and the docker-image was built and deployed to git hub. This can be seen in the packages section of git hub which is tagged with release-test

PietroConvalleAD commented:
nice, good job

NogaAD commented on README.md:
```diff
@@ -43,3 +45,24 @@ Commands for maintaining consistency and PEP8 standards across codebase, as well
+DATABASE_PASS=
+SECRET_KEY="django-insecure-x977=v5a2q-e%_5$b9ge@jzk%s_nhk2l8_use&h@*m$w33dopr"
+DJANGO_SETTINGS_MODULE=answerking.settings.base
+DATABASE_ENGINE = "django.db.backends.mysql"
```
Remove spaces around '='

JoeCSykes replied:
done

JoeCSykes committed:
Changed wait for DB file to print out red error after 20 connection tries

JoeCSykes committed:
Delete .env.production

JoeCSykes committed:
Update README.md

NogaAD committed:
Update docker-compose.yml

added extra_hosts to docker_compose to allow the use of host.docker.internal in the .env.production file

JoeCSykes committed:
changed compose file so handles errors better

JoeCSykes committed:
added more info to the print out when connection to db successful

JoeCSykes committed:
updated docker commands in README file

JoeCSykes committed:
removed unneccessary comments

JoeCSykes committed:
ran black
